src/map_object.c

```diff
@@ -107,493 +119,472 @@ LocalMapObject* sub_0205E1D0(MapObjectManager* manager, ObjectEvent* object_even
+    ObjectEvent objectEvent;
+    ObjectEvent_SetId(&objectEvent, 0);
+    ObjectEvent_SetSprite(&objectEvent, sprite);
+    ObjectEvent_SetMovement(&objectEvent, movement);
+    ObjectEvent_SetType(&objectEvent, 0);
+    ObjectEvent_SetFlagId(&objectEvent, 0);
+    ObjectEvent_SetScript(&objectEvent, 0);
+    ObjectEvent_SetFacing(&objectEvent, direction);
+    ObjectEvent_SetParam(&objectEvent, param0, 0);
+    ObjectEvent_SetParam(&objectEvent, param1, 1);
+    ObjectEvent_SetParam(&objectEvent, param2, 2);
+    ObjectEvent_SetXRange(&objectEvent, 0);
+    ObjectEvent_SetYRange(&objectEvent, 0);
+    ObjectEvent_SetXCoord(&objectEvent, x);
+    ObjectEvent_SetYCoord(&objectEvent, y);
+    ObjectEvent_SetHeight(&objectEvent, 0);
+
+    return MapObject_CreateFromObjectEvent(manager, &objectEvent, mapNo);
+}
+
+LocalMapObject *MapObject_CreateFromObjectEventWithId(MapObjectManager *manager, u16 id, u32 objectEventCount, u32 mapNo, const ObjectEvent *events) {
+    LocalMapObject *ret = NULL;
+    ObjectEvent *objectEvent = ObjectEvent_GetById(id, objectEventCount, (ObjectEvent *)events);
+    if (objectEvent != NULL) {
+        u32 flagId = ObjectEvent_GetFlagID(objectEvent);
         FieldSystem* fieldSystem = MapObjectManager_GetFieldSystemPtr(manager);
-        if (FieldSystem_FlagGet(fieldSystem, (u16)flag_id) == FALSE) {
-            ret = sub_0205E1D0(manager, event, map_no);
+        if (FieldSystem_FlagCheck(fieldSystem, (u16)flagId) == FALSE) {
+            ret = MapObject_CreateFromObjectEvent(manager, objectEvent, mapNo);
         }
     }
 
     return ret;
 }
 
-void sub_0205E38C(LocalMapObject* object, u32 sprite_id) {
-    MapObject_SetGfxID(object, sprite_id);
+void sub_0205E38C(LocalMapObject *object, u32 sprite) {
+    MapObject_SetSpriteID(object, sprite);
     sub_0205EF48(object);
     MapObject_ClearFlagsBits(object, MAPOBJECTFLAG_UNK14);
     sub_0205EFB4(object);
 }
 
-void sub_0205E3AC(LocalMapObject* object, u32 sprite_id) {
+void sub_0205E3AC(LocalMapObject *object, u32 sprite) {
     if (sub_0205F73C(object) == TRUE) {
         sub_0205E420(object);
     }
-    sub_0205E38C(object, sprite_id);
+    sub_0205E38C(object, sprite);
 }
 
-void MapObject_Remove(LocalMapObject* object) {
+void MapObject_Remove(LocalMapObject *object) {
     if (sub_0205F5D4(MapObject_GetManager(object)) == TRUE) {
         sub_0205F498(object);
     }
     sub_0205F444(object);
     sub_0205F348(object);
     sub_0205F174(sub_0205F364(object));
-    sub_0205ED80(object);
+    MapObject_Clear(object);
 }
 
-void DeleteMapObject(LocalMapObject* object) {
-    u32 flag_id = MapObject_GetFlagID(object);
-    FieldSystem* fieldSystem = MapObject_GetFieldSysPtr(object);
-    FieldSystem_FlagSet(fieldSystem, (u16)flag_id);
+void MapObject_Delete(LocalMapObject *object) {
+    u32 flagId = MapObject_GetFlagID(object);
+    FieldSystem *fieldSystem = MapObject_GetFieldSystemPtr(object);
+    FieldSystem_FlagSet(fieldSystem, (u16)flagId);
```

Comment:
does this match if you change the function signature to (FieldSystem *, u16) or change flagId to a u16?

Reply:
it's already a u16, I'm just dumb and casted unnecessarily